python/pyspark/rdd.py

```diff
@@ -2406,6 +2406,18 @@ def toLocalIterator(self):
             sock_info = self.ctx._jvm.PythonRDD.toLocalIteratorAndServe(self._jrdd.rdd())
         return _load_from_socket(sock_info, self._jrdd_deserializer)
 
+    def barrier(self):
```

Comment:
I don't know why we didn't mark the version so far here but we really should `.. versionadded:: 2.4.0` here or 

```
@since(2.4)
def barrier(self):
    ...
```